### PR TITLE
OTE_SDK. Increase code coverage. entities/metrics.py, entities/graph.py

### PR DESCRIPTION
**Pull request by saltykox**
Test list:
- ote_sdk/ote_sdk/tests/entities/test_metrics.py
- ote_sdk/ote_sdk/tests/entities/test_graph.py
- ote_sdk/ote_sdk/tests/entities/test_id.py
- ote_sdk/ote_sdk/tests/entities/interfaces/test_graph_interface.py
- ote_sdk/ote_sdk/tests/configuration/enums/test_model_lifecycle.py
- ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py

Task: CVS-72106

### CODE REVIEW

**Comment by saltykox** [comment]
run ote_sdk tests

**Comment by saltykox** [comment]
@LeonidBeynenson please review

**Comment by saltykox** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/configuration/enums/test_model_lifecycle.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_model_lifecycle(self):
+        """
+        <b>Description:</b>
+        Check ModelLifecycle Enum class elements
+
+        <b>Expected results:</b>
+        Test passes if ModelLifecycle Enum class length equal expected value and its elements have expected
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/configuration/enums/test_model_lifecycle.py]:
```suggestion
        Test passes if ModelLifecycle Enum class length is equal to expected value and its elements have expected
```

File: ote_sdk/ote_sdk/tests/entities/interfaces/test_graph_interface.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.entities.interfaces.graph_interface.IGraph.__abstractmethods__", set()
+    )
+    def test_i_graph(self):
+        """
+        <b>Description:</b>
+        Check IGraph class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/interfaces/test_graph_interface.py]:
```suggestion
        Check IGraph class object initialization
```

File: ote_sdk/ote_sdk/tests/entities/test_id.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestID:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_id(self):
+        """
+        <b>Description:</b>
+        Check ID class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_id.py]:
```suggestion
        Check ID class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.usecases.tasks.interfaces.evaluate_interface.IEvaluationTask.__abstractmethods__",
+        set(),
+    )
+    def test_evaluate_interface(self):
+        """
+        <b>Description:</b>
+        Check IEvaluationTask class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Check IEvaluationTask class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.usecases.tasks.interfaces.inference_interface.IInferenceTask.__abstractmethods__",
+        set(),
+    )
+    def test_i_inference_task(self):
+        """
+        <b>Description:</b>
+        Check IInferenceTask class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Check IInferenceTask class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.usecases.tasks.interfaces.inference_interface.IRawInference.__abstractmethods__",
+        set(),
+    )
+    def test_i_raw_inference(self):
+        """
+        <b>Description:</b>
+        Check TestIRawInference class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Check TestIRawInference class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.usecases.tasks.interfaces.optimization_interface.IOptimizationTask.__abstractmethods__",
+        set(),
+    )
+    def test_optimization_interface(self):
+        """
+        <b>Description:</b>
+        Check IOptimizationTask class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Check IOptimizationTask class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.usecases.tasks.interfaces.training_interface.ITrainingTask.__abstractmethods__",
+        set(),
+    )
+    def test_training_interface(self):
+        """
+        <b>Description:</b>
+        Check ITrainingTask class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Check ITrainingTask class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.usecases.tasks.interfaces.unload_interface.IUnload.__abstractmethods__",
+        set(),
+    )
+    def test_unload_interface(self):
+        """
+        <b>Description:</b>
+        Check IUnload class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Check IUnload class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_export_type(self):
+        """
+        <b>Description:</b>
+        Check ExportType Enum class elements
+
+        <b>Expected results:</b>
+        Test passes if ExportType Enum class length equal expected value and its elements have expected
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Test passes if ExportType Enum class length is equal to expected value and its elements have expected
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 11]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    @patch(
+        "ote_sdk.usecases.tasks.interfaces.export_interface.IExportTask.__abstractmethods__",
+        set(),
+    )
+    def test_export_interface(self):
+        """
+        <b>Description:</b>
+        Check IExportTask class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Check IExportTask class object initialization
```

File: ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py [hunk 12]
```diff
@@ -0,0 +1,297 @@
...
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_optimization_type(self):
+        """
+        <b>Description:</b>
+        Check OptimizationType Enum class elements
+
+        <b>Expected results:</b>
+        Test passes if OptimizationType Enum class length equal expected value and its elements have expected
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py]:
```suggestion
        Test passes if OptimizationType Enum class length is equal to expected value and its elements have expected
```

File: ote_sdk/ote_sdk/tests/entities/test_graph.py [hunk 13]
```diff
@@ -0,0 +1,981 @@
...
+                {"node1": "A", "node2": "D", "edge_value": 4},
+            ],
+        )
+        return directed_graph
+
+    @staticmethod
+    def compare_list_elements(actual_list: list, expected_list: list) -> None:
+        """Function to compare lists without checking elements sequence"""
+        assert len(actual_list) == len(expected_list)
+        for actual_element in actual_list:
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_graph.py]: actual_list      = [1, 1, 1, 2, 3]
expected_list = [1, 2, 3, 4, 5]

compare_list_elements(actual_list, expected_list) - > ?

File: ote_sdk/ote_sdk/tests/entities/test_graph.py [hunk 14]
```diff
@@ -0,0 +1,981 @@
...
+                expected_list=expected_attribute_dict.get("expected_value"),
+            )
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_graph_initiation(self):
+        """
+        <b>Description:</b>
+        Check Graph class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_graph.py]:
```suggestion
        Check Graph class object initialization
```

File: ote_sdk/ote_sdk/tests/entities/test_graph.py [hunk 15]
```diff
@@ -0,0 +1,981 @@
...
+
+        <b>Input data:</b>
+        Graph objects with specified "directed" and "edges" parameters
+
+        <b>Expected results:</b>
+        Test passes if Graph object "directed" attribute, "edges", "nodes" and "num_labels" properties and "num_nodes"
+        method return expected values
+
+        <b>Steps</b>
+        1. Check initiation of non-directed Graph object
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_graph.py]:
```suggestion
        1. Check initialization of non-directed Graph object
```

File: ote_sdk/ote_sdk/tests/entities/test_graph.py [hunk 16]
```diff
@@ -0,0 +1,981 @@
...
+        2. Check initiation of directed Graph object
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_graph.py]:
```suggestion
        2. Check initialization of directed Graph object
```

File: ote_sdk/ote_sdk/tests/entities/test_graph.py [hunk 17]
```diff
@@ -0,0 +1,981 @@
...
+        )
+        return multi_di_graph
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_multi_di_graph(self):
+        """
+        <b>Description:</b>
+        Check MultiDiGraph class object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_graph.py]:
```suggestion
        Check MultiDiGraph class object initialization
```

File: ote_sdk/ote_sdk/tests/entities/test_metrics.py [hunk 18]
```diff
@@ -104,3 +178,918 @@ def test_matrix_metric(self):
...
+
+        <b>Input data:</b>
+        CurveMetric object with specified name, xs and ys parameters
+
+        <b>Expected results:</b>
+        Test passes if CurveMetric object has expected attributes and methods
+
+        <b>Steps</b>
+        1. Check name, ys, xs and type attributes of CurveMetric object
+        2. Check name, ys, xs and type attributes of CurveMetric object with not specified xs initiation parameter
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_metrics.py]:
```suggestion
        2. Check name, ys, xs and type attributes of CurveMetric object with not specified xs initialization parameter
```

File: ote_sdk/ote_sdk/tests/entities/test_metrics.py [hunk 19]
```diff
@@ -104,3 +178,918 @@ def test_matrix_metric(self):
...
+        3. Check ValueError exception raised when length of ys CurveMetric object initiation parameter not equal length
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_metrics.py]:
```suggestion
        3. Check ValueError exception raised when length of ys CurveMetric object initialization parameter is not equal to length
```

File: ote_sdk/ote_sdk/tests/entities/test_metrics.py [hunk 20]
```diff
@@ -104,3 +178,918 @@ def test_matrix_metric(self):
...
+
+        <b>Expected results:</b>
+        Test passes if BarChartInfo object name, type and palette attributes and __repr__ method return expected values
+
+        <b>Steps</b>
+        1. Check name, palette and type attributes and __repr__ method values for BarChartInfo object with specified
+        palette and visualisation_type parameters
+        2. Check name, palette and type attributes and __repr__ method values for BarChartInfo object with not
+        specified palette and visualisation_type parameters
+        3. Check ValueError exception raised when visualization_type BarChartInfo object initiation parameter not equal
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_metrics.py]:
```suggestion
        3. Check ValueError exception raised when visualization_type BarChartInfo object initialization parameter is not equal
```

File: ote_sdk/ote_sdk/tests/entities/test_metrics.py [hunk 21]
```diff
@@ -72,18 +140,24 @@ def test_matrix_metric(self):
         <b>Steps</b>
         1. Create Matrices
         2. Check normalized matrices against pre-computed matrices
+        3. Check positive scenario when row_labels and column_labels parameters specified during MatrixMetric object
+        initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_metrics.py]:
```suggestion
        initialization
```

File: ote_sdk/ote_sdk/tests/entities/test_metrics.py [hunk 22]
```diff
@@ -72,18 +140,24 @@ def test_matrix_metric(self):
...
+        4. Check ValueError exception raised when row_labels parameter length not equal to number of rows of
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_metrics.py]:
```suggestion
        4. Check ValueError exception raised when row_labels parameter length is not equal to number of rows of
```

File: ote_sdk/ote_sdk/tests/entities/test_metrics.py [hunk 23]
```diff
@@ -72,18 +140,24 @@ def test_matrix_metric(self):
...
+        MatrixMetric object
+        5. Check ValueError exception raised when column_labels parameter length not equal to number of columns of
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_metrics.py]:
```suggestion
        5. Check ValueError exception raised when column_labels parameter length is not equal to number of columns of
```

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/entities/test_metrics.py (hunk 23 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_metrics.py]: Suggestions applied, thanks

**Comment by Ilya-Krylov** [comment]
Could you double check that warnings related ote_sdk testing are valid and suppress them one-by-one ?

See http://213.221.44.203/job/ote/job/precommit/job/ote_sdk_tests_runner/191/console

```
17:12:35  entities/test_graph.py: 13 warnings
17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/venv/lib/python3.8/site-packages/numpy/matrixlib/defmatrix.py:69: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
17:12:35      return matrix(data, dtype=dtype, copy=False)
17:12:35  
17:12:35  entities/test_metrics.py::TestMatrixMetricsGroup::test_matrix_metrics_group
17:12:35  entities/test_metrics.py::TestPerformance::test_performance
17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/ote_sdk/entities/metrics.py:371: RuntimeWarning: invalid value encountered in true_divide
17:12:35      self.__matrix_values = self.__matrix_values.astype(
```

**Comment by saltykox** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
As far as I can see not all suggestions have been commited AND please respond here https://github.com/openvinotoolkit/training_extensions/pull/750#discussion_r766584010

**Comment by Ilya-Krylov** [comment]
> Could you double check that warnings related ote_sdk testing are valid and suppress them one-by-one ?
> 
> See http://213.221.44.203/job/ote/job/precommit/job/ote_sdk_tests_runner/191/console
> 
> ```
> 17:12:35  entities/test_graph.py: 13 warnings
> 17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/venv/lib/python3.8/site-packages/numpy/matrixlib/defmatrix.py:69: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
> 17:12:35      return matrix(data, dtype=dtype, copy=False)
> 17:12:35  
> 17:12:35  entities/test_metrics.py::TestMatrixMetricsGroup::test_matrix_metrics_group
> 17:12:35  entities/test_metrics.py::TestPerformance::test_performance
> 17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/ote_sdk/entities/metrics.py:371: RuntimeWarning: invalid value encountered in true_divide
> 17:12:35      self.__matrix_values = self.__matrix_values.astype(
> ```

dont suppress openvino related import

File: ote_sdk/ote_sdk/tests/entities/test_graph.py (hunk 13 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_graph.py]: This function performs assertion checks, and not returns bool values

Comment (Ilya-Krylov) [reply ote_sdk/ote_sdk/tests/entities/test_graph.py]: is it ok that set(actual_list) is subser of set(expected_list) but not vice versa in general case?

Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_graph.py]: Thanks for the comment, I think it's better to use a direct comparison, I will fix incorrect comparison

**Comment by saltykox** [comment]
- fixed check_graph_list_attributes method, changed the way to check graph edges and nodes
- added suppression of "PendingDeprecationWarning" in test_graph.py and suppression of "invalid value encountered in true_divide" in test_metrics.py

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by saltykox** [comment]
run ote_sdk tests

**Comment by saltykox** [comment]
run ote_sdk tests

**Pull request merged**